torch/_inductor/fx_passes/pre_grad.py

```diff
@@ -458,3 +490,69 @@ def transpose_matmul(A: torch.Tensor, B: torch.Tensor, Atrans: bool, Btrans: boo
     if Btrans:
         B = B.transpose(-1, -2)
     return torch.matmul(A, B)
+
+
+_pointwise_ops = [
+    operator.add,
+    operator.sub,
+    operator.mul,
+    operator.iadd,
+    operator.isub,
+    operator.imul,
+    operator.itruediv,
+    operator.ifloordiv,
+]
+
+
+@register_graph_pattern(
+    CallFunction(
+        operator.setitem,
+        Arg(),
+        Arg(),
+        CallFunction(
+            _pointwise_ops,
+            CallFunction(
+                operator.getitem,
+                Arg(),
+                Arg(),
+            ),
+            Arg(),
+        ),
+    ),
+    pass_dict=split_cat_pass,
+)
+def boolean_mask_rewritten(match: Match, tensor1, index1, tensor2, index2, const_value):
```

Comment:
Basic question: this pass lives in inductor, but is it fair to say that it's basically a requirement for any compiler backend to do something similar if they want to be able to support boolean masking?

I wonder if we eventually want to add clearer/public docs and an easier way for other backends to use stuff like this. AOTAutograd is probably another example: It's not "required" for a compiler backend to use AOTAutograd, but they almost definitely want to use it if they want to properly handle autograd.

Reply:
Good point! I think eventually we should move it out of Inductor to benefit other backends. I'm curious which backends supports dynamic shapes other than Inductor?